pkg/controllers/dashboard/helm/repo_oci.go

```diff
@@ -119,11 +158,24 @@ func (o *OCIRepohandler) onClusterRepoChange(key string, clusterRepo *catalog.Cl
 				_, err := createOrUpdateMap(clusterRepo.Namespace, index, owner, o.apply)
 				if err != nil {
 					logrus.Debugf("failed to create/udpate the configmap incase of 4xx statuscode for %s", clusterRepo.Name)
-					return o.set4xxCondition(clusterRepo, errResp, originalStatus)
 				}
 			}
 
-			return o.set4xxCondition(clusterRepo, errResp, originalStatus)
+			clusterRepo.Status.NumberOfRetries++
+			if clusterRepo.Status.NumberOfRetries > retryPolicy.MaxRetry {
+				return o.set4xxCondition(clusterRepo, errResp, originalStatus)
+			}
+
+			backoff := calculateBackoff(clusterRepo, retryPolicy)
+
+			clusterRepo.Status.NextRetryAt = metav1.Time{Time: timeNow().Add(backoff)}
```

Comment:
I think it is probably unnecessary since it will likely be a couple of seconds between each retry so the user may not even have enough time to read it